lib/updateDeps.js

```diff
@@ -163,23 +163,30 @@ const resolveReleaseType = (pkg, bumpStrategy = "override", releaseStrategy = "p
 	// NOTE This fn also updates pkg deps, so it must be invoked anyway.
 	const dependentReleaseType = getDependentRelease(pkg, bumpStrategy, releaseStrategy, ignore, prefix);
 
+	// Define release type for dependent package if any of its deps changes.
+	// `patch`, `minor`, `major` — strictly declare the release type that occurs when any dependency is updated.
+	// `inherit` — applies the "highest" release of updated deps to the package.
+	// For example, if any dep has a breaking change, `major` release will be applied to the all dependants up the chain.
+	// If we want to inherit the release type from the dependent package
+	const types = ['patch', 'minor', 'major'];
+	const depIndex = dependentReleaseType ? types.indexOf(dependentReleaseType) : 4;
+	const pkgIndex = pkg._nextType? types.indexOf(pkg._nextType) : 4;
+
+	if (releaseStrategy === "inherit" && dependentReleaseType && depIndex >= pkgIndex) {
```

Comment:
We check `dependentReleaseType` here, and the ~ `if (!dependentReleaseType)` below. It seems we could move this block down.

Reply:
I think it needs to return the `dependentReleaseType` before it tries to return `pkg._nextType` or we get unintended behaviour

Reply:
Let's add a minor tweak up: replace magic number `4` with `types.length`.

Reply:
I think that's already been done?